tests/storage/hotplug.go

```diff
@@ -0,0 +1,402 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package storage
+
+import (
+	"fmt"
+	"time"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Hotplug", func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
```

Comment:
The pattern we've kind of been adopting for functional tests is that utilities are going into specialized directories, like what you've done with storage, and then the test cases themselves are all lumped into the `*_test.go` files in the `tests/` directory.

Unless this pattern has changed and I'm unaware, we should probably follow what the rest of the tests are doing and create something like a hotplug_test.go file which utilizes the helper functions you have in this storage package.

Reply:
I was following what was happening in the networking directory, and they have tests in there out of the main tests directory. From what I understood that was the direction we were trying to move to. But I have no issues making a hotplug_test.go in the main directory.

Reply:
Example of that: https://github.com/kubevirt/kubevirt/blob/master/tests/network/services.go#L175-L180

Reply:
ah, alrighty. that works for me then. lets keep it the way you have it for now and see if anyone complains. Its easy to change later on.